plottable.d.ts

```diff
@@ -1598,8 +1598,23 @@ declare module Plottable {
          * @returns The calling Component.
          */
         detach(): Component;
-        _parent(): ComponentContainer;
-        _parent(parentElement: ComponentContainer): any;
+        /**
+         * Adds a callback to be called when th Component is detach()-ed.
+         *
+         * @param {ComponentCallback} callback The callback to be added.
+         * @return {Component} The calling Component.
+         */
+        onDetach(callback: ComponentCallback): Component;
+        /**
+         * Removes a callback to be called when th Component is detach()-ed.
+         * The callback is identified by reference equality.
+         *
+         * @param {ComponentCallback} callback The callback to be removed.
+         * @return {Component} The calling Component.
+         */
+        offDetach(callback: ComponentCallback): Component;
+        parent(): ComponentContainer;
+        parent(parentElement: ComponentContainer): any;
```

Comment:
Should the above 2 be exposed?  If so, why?

Lastly, why does the second signature return an `any`